File: 724-find-pivot-index/main.go
func pivotIndex(nums []int) int {
    
    leftSum, rightSum := 0, sum(nums)

    for idx, num := range nums {
        rightSum -= num

        if rightSum == leftSum {
            return idx
        }

        leftSum += num
    }

    return -1
}


func sum(nums []int) int {
    _sum := 0

    for _, num := range nums {
        _sum += num
    }

    return _sum
}

File: 724-find-pivot-index/main.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        
        let mut left_sum: i32 = 0;
        let mut right_sum: i32 = nums.iter().sum();

        for (idx, &num) in nums.iter().enumerate() {
            
            right_sum -= num;

            if right_sum == left_sum {
                return idx as  i32;
            }

            left_sum += num;
        }

        -1
    }
}
